### PR TITLE
Add socks proxy support to APNS and health endpoint

### PR DESCRIPTION
**Pull request by PhilipGough**
ping @matzew can you review?

### CODE REVIEW

File: service/src/main/java/org/jboss/aerogear/unifiedpush/service/proxy/ProxyConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import javax.annotation.PostConstruct;
+import javax.ejb.Singleton;
+import javax.ejb.Startup;
+import java.net.Authenticator;
+import java.net.InetSocketAddress;
+import java.net.PasswordAuthentication;
+import java.net.Proxy;
+
+@Startup
+@Singleton
```
Comment (matzew) [inline service/src/main/java/org/jboss/aerogear/unifiedpush/service/proxy/ProxyConfiguration.java]: move the annotations above the `public class...` line

**Comment by matzew** [comment]
besides the one comment looks good.

One more thing, can you add the new ENVs to this document:
https://github.com/aerogear/aerogear.org/blob/master/docs/unifiedpush/ups_userguide/server-installation_properties.asciidoc#system-properties-for-the-unifiedpush-server

NOTE: there is only System properties, so feel free to create a new `Environment Variables` section _above_

File: service/src/main/java/org/jboss/aerogear/unifiedpush/service/impl/health/Ping.java [hunk 2]
```diff
@@ -26,7 +28,12 @@ public static boolean isReachable(String host, int port) {
...
         boolean reachable;
         try {
-            socket = new Socket();
+
+            if (ProxyConfiguration.hasSocks()) {
+                socket = new Socket(ProxyConfiguration.socks());
+            } else {
+                socket = new Socket();
+            }
             socket.connect(new InetSocketAddress(host, port), 2000);
```
Comment (matzew) [inline service/src/main/java/org/jboss/aerogear/unifiedpush/service/impl/health/Ping.java]: @PhilipGough is the 2 second timeout for you OK as well ?

Comment (PhilipGough) [inline service/src/main/java/org/jboss/aerogear/unifiedpush/service/impl/health/Ping.java]: Works great, returning in less than 200 typically

Comment (matzew) [inline service/src/main/java/org/jboss/aerogear/unifiedpush/service/impl/health/Ping.java]: I saw the others, that worked ;-) return in under 100 ms

**Pull request merged**